File: HW2/Bird.cpp
#include <string>

#include "Bird.h"


Bird::Bird(std::string *name, int weight, bool doesMigrate) {
    name_ = *name;
    weight_ = weight;
    doesMigrate_ = doesMigrate;
}

std::string Bird::toString() {
    std::string result = "BIRD { name = \"" + name_ + "\"; " +
                         "weight = " + std::to_string(weight_) + "; " +
                         "does_migrate = ";
    if (doesMigrate_) {
        result += "TRUE }";
    } else {
        result += "FALSE }";
    }
    return result;
}

File: HW3/Bird.py
from Creature import Creature


# The bird type of creatures.
class Bird(Creature):
    def __init__(self, name: str, weight: int, does_migrate: bool):
        super().__init__(name, weight)
        self.does_migrate = does_migrate

    def to_string(self) -> str:
        return f"BIRD {{ name = \"{self.name}\"; " \
               f"weight = {self.weight}; " \
               f"does_migrate = {self.does_migrate}; "\
               f"metric = {self.get_metric()} }}"
